### ISSUE TITLE
DKIM & SPF

### ISSUE THREAD

**Issue by nerzhul** [RainLoop/rainloop-webmail#317]
Hi,
DKIM and SPF became useful and used on the Internet.

It will be cool if rainloop could add a check when showing the mails to verify if SPF and DKIM states are correct and tell the user by a green tick that the mail comes from a verified source.

Thanks in advance

**Comment by robertol** [comment]
+1

**Comment by bsm33** [comment]
that would be really cool! +1

**Comment by Leryan** [comment]
+1

**Comment by namezk** [comment]
I think this is really the job of the MTA not RainLoop.

**Comment by bms8197** [comment]
It is the job of the MTA but in order to have that displayed for your email message in rainloop, I guess it's rainloop's job

**Comment by nerzhul** [comment]
In my request i would mean interpreting DKIM and SPF field in mail header, not testing these mechanics, it's MTA job.

**Comment by erasmuswill** [comment]
+1

**Comment by nuxwin** [comment]
Such useless thing... I could understand if you asked for SPAM flag but for SPF and DKIM signature....

**Comment by megadr01d** [comment]
I don't know if you know but this is already implemented...

**Comment by minami-o** [comment]
Indeed, in changelog of version 1.8.0 (http://www.rainloop.net/changelog/#1.8.0), dkim status is mentioned.
However, I can't find any options or settings, either in the config files or the admin panel, to enable it, and obviously it is not enabled by default, is it?
Could you please send some detail about the functionality and how to enable/see/implement it?
Thanks in advance!

17 mars 2015 14:14 "Pedro Damazio"  a écrit:
	I don't know if you know but this is already implemented...

**Comment by megadr01d** [comment]
@minami-o It's enabled by default, there's no setting for it. As long as you have DKIM set, a green checkmark will be visible next to your email. I don't know if this is by default but, to test it, you need to send an email to another account (domain). Sending a test email to another alias on your domain, won't show anything.

**Comment by minami-o** [comment]
Aha... Thanks for your help !
Indeed, it looks like my setup of DKIMProxy uses the X-Spam-Status  header entry to determine if a DKIM record is valid. This is well recognized by RoundCube, but RainLoop only recognizes the following headers at the moment:
 * Authentication-Results, whose syntax I'm not sure of, it should at least contain something like dkim=recordstatus
 * X-DKIM-Authentication-Results, which uses a simple syntax as explained in the RainLoop code: signer="domain" status="recordstatus", where domain is the signer domain and recordstatus is the status of the DKIM record, ie. pass, none or failed.
Manually changing a mail to include such header indeed resulted in a small "checked" sign next to the email address.
I'll have a look at my DKIMProxy setup to see how to use these headers instead of or in addition to the X-Spam... one. I'll post anything useful here.
Thanks again for your help, Pedro!

18 mars 2015 11:38 "Pedro Damazio"  a écrit:
	@minami-o (https://github.com/minami-o) It's enabled by default, there's no setting for it. As long as you have DKIM set, a green checkmark will be visible next to your email. I don't know if this is by default but, to test it, you need to send an email to another account (domain). Sending a test email to another alias on your domain, won't show anything.

**Comment by megadr01d** [comment]
No problem. Since I'm on shared hosting I just generated a DKIM key in cPanel, not sure how to set the headers manually.

**Comment by minami-o** [comment]
OK in the end it occurs I was using the dkim plugin provided by SPAMAssassin, which generates these X-Spam headers.
I enabled the dkimproxy.in service, and like magic, the Authentication-Results header appeared in my incoming mail.

**Comment by megadr01d** [comment]
:+1:

**Comment by ervee** [comment]
@RainLoop, can you add to the DKIM test to check for something like DKIM_SIGNED,DKIM_VERIFIED in the X-Spam-Status header to make this work with Spamassassin?

**Comment by nerzhul** [comment]
@RainLoop i think this issue is partially solved no ?

**Comment by Leryan** [comment]
@nerzhul well... we've got to get it down to one. ;-P

**Comment by tuxmartin** [comment]
Hi, can you please add SPF and DMARC status icons? 

You can get it from header`X-Spam-Status`.

```
X-Spam-Status: No, score=-0.201 tagged_above=-9999 required=5
	tests=[DKIM_SIGNED=0.1, DKIM_VALID=-0.1, DKIM_VALID_AU=-0.1,
	DKIM_VALID_EF=-0.1, HTML_MESSAGE=0.001, NO_RECEIVED=-0.001,
	NO_RELAYS=-0.001, SPF_PASS=-0.001, URIBL_BLOCKED=0.001]
```
